File: ray-js/src/Mesh.js
import Vec3 from './math/Vec3.js';
module.exports = class Mesh {
	constructor(geometry, material) {
		this.id = RAY.objectId;
		RAY.objectId += 1;
		this.geometry = geometry;
		this.material = material;
		this.position = new Vec3();
	}
}

File: helios/src/Mesh.cpp
using namespace helios;
using namespace std;

Mesh::Mesh(Geometry* geometry, Material* material) {
	this->geometry = geometry;
	this->material = material;
	this->position = new Vector3(0, 0, 0);
	this->scale = new Vector3(1, 1, 1);
	this->orientation = Matrix<float>::identity(3);
}
